Workiva/wGulp
# Move require statements into function scope to improve startup time

## Problem

wGulp startup time has been getting worse and now it is noticeably annoying clocking in between 2.5 seconds and 3.5 seconds. This is time before any task starts running and happens _every time_ you run a `gulp` command using wGulp. Here is an example:

``` bash
merge_options: 3ms
detect-cycles: 8ms
require-subtasks: 2177ms
create-subtask-tasks: 1ms
require-run-sequence: 2ms
build-bundle-tasks: 505ms
require-task-folders: 890ms
----
total-startup: 3588ms
```
## Solution

By moving require calls into task functions we can prevent node.js from following the whole dependency tree with a single thread on startup. If `require('browserify');` is at the very top of a source file, for example, node.js will follow that statement into the browserify module, which requires a bunch of other things. It does this all synchronously and in one thread.

If we delay that to happen within the gulp tasks, it can all happen in parallel/asynchronously (since gulp is asynchronous in nature). We also get through the startup/setup of wGulp a **lot** faster.

Furthermore, you generally aren't running every single task within wGulp on a build or test run, so you shouldn't have to spend time requiring task deps that you aren't using for that particular task run. This accomplishes that as well.

Here are a couple of runs to give you an idea of the new startup time:

``` bash
merge_options: 5ms
detect-cycles: 9ms
require-subtasks: 8ms
create-subtask-tasks: 1ms
require-run-sequence: 0ms
build-bundle-tasks: 52ms
require-task-folders: 50ms
-----------
total-startup: 128ms
```

``` bash
merge_options: 5ms
detect-cycles: 8ms
require-subtasks: 13ms
create-subtask-tasks: 2ms
require-run-sequence: 1ms
build-bundle-tasks: 90ms
require-task-folders: 72ms
-----------
total-startup: 193ms
```
## Testing
- Ensure tests pass
- Ensure examples work
- Try it in some wGulp projects

@trentgrover-wf 
@evanweible-wf 
@shanesizer-wf
